ddtrace/contrib/fastapi/patch.py

```diff
@@ -35,12 +37,39 @@ def traced_init(wrapped, instance, args, kwargs):
     wrapped(*args, **kwargs)
 
 
+async def traced_serialize_response(wrapped, instance, args, kwargs):
```

Comment:
This function seems like a generic async wrapper. Couldn't we combine a generic form of this with `ddtrace.contrib.trace_utils.with_traced_module` and use it here? 

That's not a blocker for me but in that case we would like to make this function private so it can be replaced down the road.

Reply:
yes, @Kyle-Verhoog has an `async_with_module` as part of #3362